src/darwin/Framework/CHIPTests/MTRXPCListenerSampleTests.m

```diff
@@ -1865,6 +1887,30 @@ - (void)test015_MTRDeviceInteraction
         }, @(NO));
 }
 
+- (void)test016_DownloadLog
+{
+    XCTestExpectation * expectation =
+        [self expectationWithDescription:@"Download EndUserSupport log"];
+
+    MTRBaseDevice * device = GetConnectedDevice();
+    dispatch_queue_t queue = dispatch_get_main_queue();
+
+    [device downloadLogOfType:MTRDiagnosticLogTypeEndUserSupport
+                      timeout:10
+                        queue:queue
+                   completion:^(NSURL * _Nullable url, NSError * _Nullable error) {
+                       NSLog(@"downloadLogOfType: url: %@, error: %@", url, error);
+                       XCTAssertNil(error);
+
+                       NSError * readError;
+                       NSString * fileContent = [NSString stringWithContentsOfURL:url encoding:NSUTF8StringEncoding error:&readError];
+                       XCTAssertNil(readError);
+                       XCTAssertTrue([fileContent isEqualToString:@"This is a simple log\n"]);
```

Comment:
Please use `XCTAssertEqualObjects` for equality assertions (it provides a useful error message on failure)